cmd/pkl-gen-go/pkl-gen-go.go

```diff
@@ -29,247 +32,357 @@ import (
+
+When using a base path, any package that does not belong to the path will be skipped from
+code generation.
+`
+
+// root cobra command for pkl-gen-go
+var command = cobra.Command{
+	Use:     "pkl-gen-go [flags] <module>",
+	Short:   "Generates Go bindings for a Pkl module",
+	Long:    longDescription,
+	PreRunE: commandPreRunE,
+	RunE:    commandRunE,
 }
 
-func newEvaluator() (pkl.Evaluator, error) {
-	projectDirFlag := ""
-	if settings.ProjectDir != nil {
-		if filepath.IsAbs(*settings.ProjectDir) {
-			projectDirFlag = *settings.ProjectDir
-		} else {
-			settingsUri, err := url.Parse(settings.Uri)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse settings.pkl URI: %w", err)
-			}
-			projectDirFlag = path.Join(settingsUri.Path, "..", *settings.ProjectDir)
-		}
+// flag names
+const (
+	flagNamePrintVersion      = "version"
+	flagNameBasePath          = "base-path"
+	flagNameGeneratorSettings = "generator-settings"
+	flagNameGenerateScript    = "generate-script"
+	flagNameProjectDirname    = "project-dir"
+	flagNameSuppressWarnings  = "suppress-format-warning"
+	flagNameOutputPath        = "output-path"
+	flagNamePackageMappings   = "mapping"
+	flagNameAllowedModules    = "allowed-modules"
+	flagNameAllowedResources  = "allowed-resources"
+	flagNameDryRun            = "dry-run"
+)
+
+// initialize command flag set
+func init() {
+	// find the current working directory
+	cwd, cwdErr := os.Getwd()
+	handleFatalError(cwdErr)
+
+	// define root command flags
+	flagSet := command.Flags()
+	flagSet.Bool(flagNamePrintVersion, false, "Print the version and exit")
+	flagSet.String(flagNameBasePath, "", "The base path used to determine relative output")
+	flagSet.String(flagNameGeneratorSettings, "", "path to a generator settings file")
+	flagSet.String(flagNameGenerateScript, "", "The Generate.pkl script to use")
+	flagSet.String(flagNameProjectDirname, "", "project directory from which dependency and evaluator settings are loaded")
+	flagSet.Bool(flagNameSuppressWarnings, false, "Suppress warnings around formatting issues")
+	flagSet.String(flagNameOutputPath, cwd, "The output directory to write generated sources into")
+	flagSet.StringToString(flagNamePackageMappings, nil, "The mapping of a Pkl module name to a Go package name")
+	flagSet.StringSlice(flagNameAllowedModules, nil, "URI patterns that determine which modules can be loaded and evaluated")
+	flagSet.StringSlice(flagNameAllowedResources, nil, "URI patterns that determine which resources can be loaded and evaluated")
+	flagSet.Bool(flagNameDryRun, false, "Print out the names of the files that will be generated, but don't write any files")
+}
+
+// context keys
+type ckGeneratorSettings struct{}
+type ckSuppressWarnings struct{}
+type ckOutputPath struct{}
+
+// command pre-run logic
+func commandPreRunE(cmd *cobra.Command, args []string) error {
+	flagSet := cmd.Flags()
+
+	// resolve flag values
+	printVersion := unwrapValue(flagSet.GetBool(flagNamePrintVersion))
```

Comment:
This bit is where something more like what exists in https://github.com/apple/pkl-go/pull/53 would be useful. Instead of eating the error with `unwrapValue`, we could have a constructor that creates a structure with all the CLI flag values in it that returns an error that we can handle properly.